src/rust/engine/src/nodes.rs

```diff
@@ -554,22 +561,29 @@ impl Node for ExecuteProcess {
   fn run(self, context: Context) -> NodeFuture<ProcessResult> {
     let request = self.0;
     let context2 = context.clone();
+
+    let command_runner = process_execution::local::CommandRunner::new(
+      context.core.store.clone(),
+      context.core.fs_pool.clone(),
+    );
+
     // TODO: Process pool management should likely move into the `process_execution` crate, which
     // will have different strategies depending on remote/local execution.
     context
       .core
-      .pool
+      .fs_pool
       .spawn_fn(move || {
         let tmpdir = TempDir::new("process-execution").unwrap();
+        let dir = tmpdir.path().to_owned();
         context2
           .core
           .store
-          .materialize_directory(tmpdir.path().to_owned(), request.input_files)
-          .map(move |_| {
-            ProcessResult(
-              process_execution::local::run_command_locally(request, tmpdir.path()).unwrap(),
-            )
-          })
+          .materialize_directory(dir.clone(), request.input_files)
+          .and_then(move |()| command_runner.run(request, &dir))
+          // Force tmpdir not to get dropped until the command is finished being run, because it's
```

Comment:
Would it be worthwhile to have `run_command_locally` explicitly borrow a `TempDir` to enforce this? Will we ever be using this outside a tmpdir?

Reply:
I can believe a future world where we're e.g. using a FUSE filesystem rather than a tmpdir, but until we actually have other constraints this is a great suggestion! Done.